source/FreeRTOS_Stream_Buffer.c

```diff
@@ -281,20 +256,21 @@ size_t uxStreamBufferAdd( StreamBuffer_t * pxBuffer,
 
     /* The number of bytes that can be written is the minimum of the number of
      * bytes requested and the number available. */
-    uxCount = FreeRTOS_min_size_t( uxSpace, uxCount );
+    uxByteCount = FreeRTOS_min_size_t( uxSpace, uxByteCount );
```

Comment:
MISRA doesn't like when we change the value of a parameter, like here in :
~~~
    uxByteCount = FreeRTOS_min_size_t( uxSpace, uxByteCount );
~~~
That is why the help variable `uxCount` was introduced.

Reply:
Yeah just to add, it violates MISRA 17.8 A function parameters should not be modified. 
So it is expected to copy the param to local object and modify it. Hence uxCount was introduced.